cmake/TargetFlags.cmake

```diff
@@ -48,3 +48,12 @@ function(target_flags FLAGS TARGET)
     # message("_flags: ${_flags}")
     set(${FLAGS} ${_flags} PARENT_SCOPE)
 endfunction()
+
+function(target_compile_flags FLAGS TARGET)
+    set(_flags)
+    append_flags(_flags ${TARGET} "INTERFACE_COMPILE_OPTIONS" "")
+    append_flags(_flags ${TARGET} "INTERFACE_COMPILE_DEFINITIONS" "-D")
+    append_flags(_flags ${TARGET} "INTERFACE_INCLUDE_DIRECTORIES" "-isystem ")
```

Comment:
This wont get the complete flags as it wont transitively traverse the targets, so it will miss the flags from `hip::host`.

Reply:
Fixed